src/fileioc/fileioc.asm

```diff
@@ -1379,21 +1362,60 @@ ti_RclVar:
 
 ;-------------------------------------------------------------------------------
 ti_ArchiveHasRoom:
-; checks if there is room in the archive before a garbage collect
+; checks if there is room in the archive without triggering a garbage collect.
 ; args:
 ;  sp + 3 : number of bytes to store into the archive
 ; return:
 ;  true if there is room, false if not
 	pop	de
 	ex	(sp),hl
 	push	de
-util_ArchiveHasRoom:
+.entry:
+	ex.s	de,hl
+	push	de
+	pop	bc
+	call	ti.FindFreeArcSpot
+	ld	a,1
+	ret	nz
+	xor	a,a
+	ret
+
+;-------------------------------------------------------------------------------
+ti_ArchiveHasRoomVar:
+; checks if there is room in the archive without triggering a garbage collect.
+; args:
+;  sp + 3 : handle to variable
+; return:
+;  true if there is room, false if not
+	pop	de
+	pop	bc
+	push	bc
+	push	de
+	call	util_is_slot_open
+	jr	nz,.fail
+	call	util_get_vat_ptr
+	ld	hl,(hl)
+	ld	bc,-6
+	add	hl,bc
+	ld	c,(hl)			; get var size
+	call	util_get_data_ptr
+	ld	hl,(hl)
```

Comment:
This should be ld de,(hl) to match the chkfindsym outputs as used below

Reply:
yep